app/models/custom_button.rb

```diff
@@ -143,4 +146,19 @@ def self.get_user(user)
     raise _("Unable to find user '%{user}'") % {:user => user} if user.nil?
     user
   end
+
+  def clone(options = {})
```

Comment:
Is this purposely an override method or simply a new method? If it is an override, `Object#clone` does not take options. If it is a new method consider `deep_copy` like `Dialog#deep_copy` we introduced for this project.

Same reasoning for `CustomButtonSet#deep_clone`.